### PR TITLE
BHBC-1201: Push project/survey metadata (EML) to backbone on project/survey create/update

### PR DESCRIPTION
**Pull request by NickPhura**
# Overview

## Links to Jira tickets

https://quartech.atlassian.net/browse/BHBC-1201

## Description of relevant changes

Add metadata submission function to build DwCA archive with just EML meta, and submit to the backbone.
 - Also preemptively added a function to build the full DwCA archive and submit it, though it currently doesn't fetch/include the occurrence data, as that will be part of a future ticket.  

Call metadata submission function from all endpoints that involve creating or update projects or surveys.  

Added new env var so we can update the backbone intake path, if it changes.
Added new env var so we can disable/enable submitting to the backbone (defaulting to false currently)
 - Will need update your `.env`

## Testing
You will need to run both SIMS and Platform at the same time. You will have to make some changes to one or both of their `.env` files.
- There are a few env vars in SIMS that need to be populated (towards the top, they reference `backbone`).
- You will need to change the env vars mentioned in the comment block at the top of `env.docker`, mainly ports, so that SIMS and Platform dont reference the same ports. 

## PR Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### Code

- [ ] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [ ] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [ ] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [ ] My code changes account for null/undefined values and handle errors appropriately
- [ ] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [ ] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [ ] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [ ] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

### CODE REVIEW

File: api/openshift/api.dc.yaml [hunk 1]
```diff
@@ -25,7 +25,14 @@ parameters:
...
     value: ''
   - name: BACKBONE_API_HOST
+    required: true
     description: API host for BioHub Platform Backbone
+  - name: BACKBONE_INTAKE_PATH
+    required: true
+    description: API path for BioHub Platform Backbone DwCA submission intake endpoint
+  - name: BACKBONE_INTAKE_ENABLED
+    required: true
+    description: Controls whether or not SIMS will submit DwCA datasets to teh BioHub Platform Backbone
```
Comment (curtisupshall) [inline api/openshift/api.dc.yaml]: Spelling error

File: api/src/paths/project/{projectId}/funding-sources/{pfsId}/delete.ts [hunk 2]
```diff
@@ -32,13 +33,14 @@ DELETE.apiDoc = deleteFundingSourceApiDocObject(
 
 export function deleteFundingSource(): RequestHandler {
   return async (req, res) => {
-    defaultLog.debug({ label: 'Delete project funding source', message: 'params', req_params: req.params });
+    const projectId = Number(req.params.projectId);
+    const pfsId = Number(req.params.pfsId);
```
Comment (curtisupshall) [inline api/src/paths/project/{projectId}/funding-sources/{pfsId}/delete.ts:37]: Is this shorthand for Platform Service ID?

**Review by curtisupshall — DISMISSED** [review]
It looks great! 🏝️

File: api/src/paths/project/{projectId}/funding-sources/{pfsId}/delete.ts (hunk 2 above)
Comment (NickPhura) [reply api/src/paths/project/{projectId}/funding-sources/{pfsId}/delete.ts:37]: It stands for `project funding source id`. Some of this code could use a bit of a refactor/cleanup still, but I was just touching the bits that directly related to my changes.

File: api/openshift/api.dc.yaml (hunk 1 above)
Comment (NickPhura) [reply api/openshift/api.dc.yaml]: Fixed in latest commit

**Review by curtisupshall — DISMISSED** [review]
⛱️

**Review by curtisupshall — APPROVED** [review]
Pulled down and tested a few scenarios. Looks great!

**Pull request merged**